packages/core/src/utils/__tests__/indyProofRequest.test.ts

```diff
@@ -0,0 +1,90 @@
+import { checkProofRequestForDuplicates } from '..'
+
+import {
+  AriesFrameworkError,
+  AttributeFilter,
+  PredicateType,
+  ProofAttributeInfo,
+  ProofPredicateInfo,
+  ProofRequest,
+} from '@aries-framework/core'
+
+export const INDY_CREDENTIAL_OFFER_ATTACHMENT_ID = 'libindy-cred-offer-0'
+
+describe('Present Proof', () => {
+  let credDefId: string
+
+  beforeAll(async () => {
+    credDefId = '9vPXgSpQJPkJEALbLXueBp:3:CL:57753:tag1'
+  })
+
+  test('attribute names match, same cred def filter', async () => {
+    const attributes = {
+      name: new ProofAttributeInfo({
+        name: 'age',
+        restrictions: [
+          new AttributeFilter({
+            credentialDefinitionId: credDefId,
+          }),
+        ],
+      }),
+      age: new ProofAttributeInfo({
+        name: 'age',
+        restrictions: [
+          new AttributeFilter({
+            credentialDefinitionId: credDefId,
+          }),
+        ],
+      }),
+    }
+
+    const nonce = 'testtesttest12345'
```

Comment:
@TimoGlastra I created an issue for this, #646. Is this solution fine or do we need to expose the generateNonce function in a more accessible way?